eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubSender.java

```diff
@@ -0,0 +1,245 @@
+                    logger.asWarning().log(String.format("[%s], size: %s, total: %s", batch.partitionKey(), batch.size(), totals));
+                });
+            })
+            .doOnError(error -> {
+                logger.asError().log(error.toString());
+            })
+            .doOnComplete(() -> {
+                logger.asInformational().log(String.format("TOTAL BATCHES: %s. EVENTS: %s", number.get(), totalEvents.get()));
+            })
+            .then();
+    }
+
+    /**
+     * Sends the {@code events} to the specified Event Hubs {@code partitionId}. If {@link EventData#partitionKey()} is
+     * specified, these are dropped from being sent.
+     *
+     * @param partitionId Event Hubs partition to send the events to.
+     * @param events Events to send to the service.
+     * @return A {@link Mono} that completes when all events are pushed to the service.
+     */
+    public Mono<Void> send(String partitionId, Publisher<EventData> events) {
+        Objects.requireNonNull(partitionId);
+        Objects.requireNonNull(events);
+
+        //TODO (conniey): What happens when someone does specify event.partitionKey()? Should I just drop them or send
+        // them in another batch?
+        final Flux<EventData> filtered = Flux.defer(() -> events)
+            .filter(event -> event.partitionKey() == null);
+
+        return send(partitionId, filtered);
+    }
+
+    private Mono<Void> send(String partitionId, Flux<EventData> events)  {
+        final AtomicInteger number = new AtomicInteger(0);
+        final AtomicInteger totalEvents = new AtomicInteger(0);
+
+        return Flux.defer(() -> events)
+            .windowTimeout(100, Duration.ofMillis(500))
+            .flatMap(flux -> flux.groupBy(event -> {
+                if (partitionId != null) {
+                    return partitionId;
+                }
+
+                return event.partitionKey() != null ? event.partitionKey() : EMPTY_PARTITION_KEY;
+            }))
+            .flatMap(group -> {
+                final EventDataCollector eventDataCollector = EMPTY_PARTITION_KEY.equals(group.key())
+                    ? new EventDataCollector(maxMessageSize, null)
+                    : new EventDataCollector(maxMessageSize, group.key());
+
+                return group.collect(eventDataCollector);
+            })
+            //TODO (conniey): Emit contents into Event Hubs.
+            .doOnNext(batchList -> {
+                batchList.forEach(batch -> {
+                    number.incrementAndGet();
+                    final int totals = totalEvents.addAndGet(batch.size());
+                    logger.asWarning().log(String.format("[%s], size: %s, total: %s", batch.partitionKey(), batch.size(), totals));
+                });
+            })
+            .doOnError(error -> {
+                logger.asError().log(error.toString());
+            })
+            .doOnComplete(() -> {
```

Comment:
Code style preference here for me is to have the new operation at the end of the previous line, rather than on a new line.